iree/compiler/Conversion/LinalgToLinalg/Conv2D1x1ToMatmul.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mlir/Dialect/Linalg/IR/LinalgOps.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace {
+
+// Converts linalg.conv_2d_input_nhwc_filter_nhwc op to linalg.matmul
+class Convert1x1ConvolutionMatmulOp
+    : public OpRewritePattern<linalg::ConvInputNHWCFilterHWCFOp> {
+ public:
+  using OpRewritePattern<linalg::ConvInputNHWCFilterHWCFOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::ConvInputNHWCFilterHWCFOp convOp,
+                                PatternRewriter &rewriter) const override {
+    ShapedType inputShapeType = convOp.getInputShapedType(0);
+    ShapedType filterShapeType = convOp.getInputShapedType(1);
+    ShapedType outputShapeType = convOp.getOutputShapedType(0);
+
+    auto inputShape = inputShapeType.getShape();
+    auto filterShape = filterShapeType.getShape();
+    auto outputShape = outputShapeType.getShape();
+
+    if (filterShape[0] != 1 || filterShape[1] != 1) return failure();
+
+    // TODO(ataei): Support conversion to linalg.batch_matmul.
+    if (inputShape[0] != 1) return failure();
+
+    if (!llvm::all_of(convOp.strides(), [](APInt element) {
+          return element.getSExtValue() == 1;
+        }))
+      return failure();
+    if (!llvm::all_of(convOp.dilations(), [](APInt element) {
+          return element.getSExtValue() == 1;
+        }))
+      return failure();
+
+    // Gets an affine dim expression.
+    auto dim = [&](int i) -> AffineExpr {
+      return rewriter.getAffineDimExpr(i);
+    };
+    SmallVector<linalg::ReassociationExprs, 4> reassociationMaps = {
```

Comment:
You can use `SmallVector<ReassociationIndices>` instead and not have to deal with `AffineExpr`

Reply:
Thanks! that's better :)